internal/advertisement-operator/broadcaster.go

```diff
@@ -354,16 +354,14 @@ func (b *AdvertisementBroadcaster) SendSecretToForeignCluster(secret *corev1.Sec
 
 func (b *AdvertisementBroadcaster) NotifyAdvertisementDeletion() error {
 	advName := pkg.AdvertisementPrefix + b.HomeClusterId
-	obj, err := b.RemoteClient.Resource("advertisements").Get(advName, metav1.GetOptions{})
+	_, err := b.RemoteClient.Resource("advertisements").Get(advName, metav1.GetOptions{})
```

Comment:
I think this `Get` is not needed anymore since you only need the advName to perform deletion: if the adv doesn't exist, `Delete` will return this error